src/sequence/mod.rs

```diff
@@ -719,22 +723,162 @@ mod tests {
+
+    /// Generate a vec of Sequence replicas of some length
+    fn generate_replicas(quantity: i32, xorname: XorName, tag: u64) -> Result<Vec<Sequence>> {
+        let mut replicas: Vec<Sequence> = vec![];
+        let owner = gen_public_key();
+
+        for _ in 0..quantity {
+            let mut seq = Sequence::new_public(owner, xorname, tag);
+            let perms = BTreeMap::default();
+            let owner_op = seq.set_public_policy(owner, perms)?;
+            seq.apply_public_policy_op(owner_op)?;
+            replicas.push(seq)
+        }
+
+        Ok(replicas)
+    }
+
+    proptest! {
+        #[test]
+        fn proptest_seq_doesnt_crash_with_random_data(s in "\\PC*".prop_map(|s| s.into_bytes())) {
+            let actor1 = gen_public_key();
+            let sequence_name = XorName::random();
+
+            let sdata_tag = 43_001u64;
+
+            // Instantiate the same Sequence on two replicas with two diff actors
+            let mut replica1 = Sequence::new_public(actor1, sequence_name, sdata_tag);
+            let mut replica2 = Sequence::new_public(actor1, sequence_name, sdata_tag);
+
+            // Set Actor1 as the owner
+            let perms = BTreeMap::default();
+            let owner_op = replica1.set_public_policy(actor1, perms)?;
+            replica2.apply_public_policy_op(owner_op)?;
+
+            // Append an item on replicas
+            let append_op1 = replica1.append(s.clone())?;
+            let append_op2 = replica2.append(s)?;
+
+            replica1.apply_data_op(append_op1)?;
+            replica2.apply_data_op(append_op2)?;
+
+            verify_data_convergence(vec![replica1, replica2], 1);
+
+        }
+
+
+        #[test]
+        fn proptest_seq_converge_with_many_random_data( dataset in prop::collection::vec("\\PC*", 1..1000) ) {
+
+            let actor1 = gen_public_key();
+            let sequence_name = XorName::random();
+
+            let sdata_tag = 43_001u64;
+
+            // Instantiate the same Sequence on two replicas with two diff actors
+            let mut replica1 = Sequence::new_public(actor1, sequence_name, sdata_tag);
+            let mut replica2 = Sequence::new_public(actor1, sequence_name, sdata_tag);
+
+            // Set Actor1 as the owner
+            let perms = BTreeMap::default();
+            let owner_op = replica1.set_public_policy(actor1, perms)?;
+            replica2.apply_public_policy_op(owner_op)?;
+
+            let dataset_length = dataset.len() as u64;
+
+            // insert our data at replicas
+            for data in dataset {
+                let data = data.into_bytes();
+                // Append an item on replica1
+                let append_op1 = replica1.append(data.clone())?;
+                let append_op2 = replica2.append(data)?;
+
+                replica1.apply_data_op(append_op1)?;
+                replica2.apply_data_op(append_op2)?;
+            }
+
+            verify_data_convergence(vec![replica1, replica2], dataset_length);
```

Comment:
Same comment as in previous test about this check, we are having a false positive testing convergence here if I'm correct.